airbyte-config/config-models/src/main/java/io/airbyte/config/helpers/StateMessageHelper.java

```diff
@@ -39,14 +65,10 @@ public static Optional<StateWrapper> getTypedState(final JsonNode state) {
         return Optional.of(getLegacyStateWrapper(state));
       }
       if (stateMessages.size() == 1 && stateMessages.get(0).getType() == AirbyteStateType.GLOBAL) {
-        return Optional.of(new StateWrapper()
-            .withStateType(StateType.GLOBAL)
-            .withGlobal(stateMessages.get(0)));
+        return Optional.of(provideGlobalState.apply(stateMessages, useStreamCapableState));
```

Comment:
Since the validation on `stateMessages` size and type is being performed in this method, I think it would be cleaner if this method passed `stateMessages.get(0)` into `provideGlobalState`, so that `provideGlobalState` doesn't need to make any assumptions about the `stateMessages` array.